source/FileCache.cs

```diff
@@ -115,6 +108,10 @@ public static async Task<MemoryStream> HitAsync(string url)
             {
                 return null;
             }
+            catch(Exception ex)
```

Comment:
generally, it is not a good idea to catch `Exception` - since it will swallow all errors in the block. Is there a more specific exception type we can catch here?